app/src/main/java/com/example/harvestfresh/fragments/EventsFragment.java

```diff
@@ -79,154 +102,169 @@ public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceStat
+                }
+                else{
+                    Toast.makeText(getContext(), "Permission Failed", Toast.LENGTH_SHORT).show();
+                }
+        }
+    }
+
+    private void getFinalLocation() {
+        //one thing i missed in permission let's complete it
+        try{
+            if(isGpsLocation){
+                locationManager.requestLocationUpdates(LocationManager.GPS_PROVIDER,1000*60*1,10, (LocationListener) getContext());
+                if(locationManager!=null){
+                    loc=locationManager.getLastKnownLocation(LocationManager.GPS_PROVIDER);
+                    if(loc!=null){
+                        updateUi(loc);
+                    }
+                }
+            }
+            else if(isNetworklocation){
+                locationManager.requestLocationUpdates(LocationManager.NETWORK_PROVIDER,1000*60*1,10, (LocationListener) getContext());
+                if(locationManager!=null){
+                    loc=locationManager.getLastKnownLocation(LocationManager.NETWORK_PROVIDER);
+                    if(loc!=null){
+                        updateUi(loc);
+                    }
+                }
+            }
+            else{
+                Toast.makeText(getContext(), "Can't Get Location", Toast.LENGTH_SHORT).show();
+            }
+
+        }catch (SecurityException e){
+            Toast.makeText(getContext(), "Can't Get Location", Toast.LENGTH_SHORT).show();
+        }
+
+    }
+
+    private void updateUi(Location loc) {
+        if(loc.getLatitude()==0 && loc.getLongitude()==0){
+            getDeviceLocation();
+        }
+        else{
+            progressDialog.dismiss();
+            LatLng latLng=new LatLng(loc.getLatitude(),loc.getLongitude());
+            map.addMarker(new MarkerOptions().position(latLng).title("Current Location"));
+            map.moveCamera(CameraUpdateFactory.newLatLng(latLng));
+            map.animateCamera(CameraUpdateFactory.newLatLngZoom(latLng,15f));
+
+        }
+
+    }
+
+    private void showSettingForLocation() {
+        AlertDialog.Builder al=new AlertDialog.Builder(getContext());
+        al.setTitle("Location Not Enabled!");
+        al.setMessage("Enable Location ?");
+        al.setPositiveButton("Yes", new DialogInterface.OnClickListener() {
+            @Override
+            public void onClick(DialogInterface dialog, int which) {
+                Intent intent=new Intent(Settings.ACTION_LOCATION_SOURCE_SETTINGS);
+                startActivity(intent);
+            }
+        });
+        al.setNegativeButton("No", new DialogInterface.OnClickListener() {
+            @Override
+            public void onClick(DialogInterface dialog, int which) {
+                dialog.cancel();
+            }
+        });
+        al.show();
+    }
+
+
+    public void onLocationChanged(Location location) {
+        updateUi(location);
+    }
+
+    public void onStatusChanged(String provider, int status, Bundle extras) {
+
```

Comment:
generallly, we try to avoid stub methods even as temporary, but if theyre necessary, a comment //TODO: explaining what youre going to do with them, and, in java, possibly throwing a NotImplementedException() makes it clear what their purpose is.